Assets/Scripts/Tools/Editor/DeveloperControls.cs

```diff
@@ -0,0 +1,242 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class DeveloperControls : EditorWindow
+{
+    // an array that contains all the possible commands
+    string[] commandArray = new string[]
+    {
+        "help", "killable", "kill", "restart", "spawn"
+    };
+
+    // stuff that the commands modify
+    GameObject player;
+
+    // for formatting the gui
+    Vector2 scrollPos;
+    float windowWidth;
+    float windowHeight;
+
+    // deals with taking in user input and the console log
+    string userInput = "Enter Command Here";
+    string[] inputModified;
+    string consoleLog = "";
+
+    // Open the Editor Window
+    [MenuItem("Tools/Developer Console")]
+    public static void ShowWindow()
+    {
+        EditorWindow.GetWindow<DeveloperControls>();
+    }
+
+    private void OnInspectorUpdate() // To repaint the window
+    {
+        Repaint();
+    }
+
+    private void OnEnable()
+    {
+        player = GameObject.Find("Player"); // finds the player's game object for modification
```

Comment:
I would recommend extracting this finding player code and turning it into its own function because you use it multiple times in this script. Also I would change how you find the player object because using its object name can be volitile, instead maybe try using tags instead.